target/linux/ipq40xx/image/generic.mk

```diff
@@ -217,6 +221,27 @@ define Device/asus_rt-ac58u
 endef
 TARGET_DEVICES += asus_rt-ac58u
 
+define Device/asus_rt-acrh17
+	$(call Device/FitImage)
+	DEVICE_VENDOR := ASUS
+	DEVICE_MODEL := RT-ACRH17
+	SOC := qcom-ipq4019
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	DTB_SIZE := 65536
+	IMAGE_SIZE := 20439364
+	FILESYSTEMS := squashfs
+#	RT-AC82U is nowhere to be found online
+#	Rather, this device is a/k/a RT-AC42U
+#	But we'll go with what the vendor firmware has...
+	UIMAGE_NAME:=$(shell echo -e '\03\01\01\01RT-AC82U')
+	UIMAGE_MAGIC := 0x27051956
+	IMAGES += factory.trx
+	IMAGE/factory.trx := copy-file $(KDIR)/tmp/$$(KERNEL_INITRAMFS_IMAGE) | uImage none
```

Comment:
This was tried before, but the Initramfs images is not available in the ImageBuilder. This sadly breaks services like OpenWrt's own firmware-selector https://firmware-selector.openwrt.org/ and third-party https://chef.libremesh.org/

Reply:
> This was tried before, but the Initramfs images is not available in the ImageBuilder. This sadly breaks services like OpenWrt's own firmware-selector https://firmware-selector.openwrt.org/ and third-party https://chef.libremesh.org/

Too bad. :(
The same can be accomplished by `KERNEL_INITRAMFS` and `KERNEL_INITRAMFS_SUFFIX` I guess?

Reply:
The Initramfs is useful for the Bootloader's SDRAM boot option. I would hate to see that being rebranded as a "factory.trx" when it's not doing a proper installation like a factory image would. (sysupgrade now also complains about when you try to sysupgrade from a initramfs).